wallet-test/src/test/scala/org/bitcoins/wallet/WalletUnitTest.scala

```diff
@@ -136,16 +136,26 @@ class WalletUnitTest extends BitcoinSWalletTest {
 
   it should "fail to unlock the wallet with a bad password" in {
     wallet: Wallet =>
-      val badpassphrase = AesPassword.fromNonEmptyString("bad")
+      val badPassphrase = Some(AesPassword.fromNonEmptyString("bad"))
 
-      val errorType = wallet.unlock(badpassphrase, None) match {
+      val errorType = wallet.unlock(badPassphrase, None) match {
         case Right(_)  => fail("Unlocked wallet with bad password!")
         case Left(err) => err
       }
       errorType match {
-        case KeyManagerUnlockError.MnemonicNotFound          => fail(MnemonicNotFound)
-        case KeyManagerUnlockError.BadPassword               => succeed
-        case KeyManagerUnlockError.JsonParsingError(message) => fail(message)
+        case KeyManagerUnlockError.MnemonicNotFound => fail(MnemonicNotFound)
+        case KeyManagerUnlockError.BadPassword      =>
+          // If wallet is unencrypted then we shouldn't get a bad password error
+          wallet.walletConfig.aesPasswordOpt match {
+            case Some(_) => succeed
+            case None    => fail()
+          }
+        case KeyManagerUnlockError.JsonParsingError(message) =>
+          // If wallet is encrypted then we shouldn't get a json parsing error
```

Comment:
same thing applies here, `... shouldn't get a json parsing error` when we are pattern matching for the `JsonParsingError`

Reply:
We shouldn't get a `JsonParsingError` error when the wallet uses an aes password. But if we don't have an aes password then we should get the error, the match on the aesPasswordOpt handles this